tests/test_table.py

```diff
@@ -1498,6 +1498,103 @@ def test_add_column(self, test_tbl: catalog.Table) -> None:
         t = pxt.get_table(t._name)
         assert len(t.columns) == num_orig_cols
 
+    def test_bool_column(self, reset_db: None, reload_tester: ReloadTester) -> None:
+        # test adding a bool column with constant value
+        t1 = pxt.create_table('test1', {'c1': pxt.Int})
+        t1.insert([{'c1': 1}, {'c1': 2}])
+        assert t1.count() == 2
+        t1.add_column(bool_const=False)
+        assert t1.where(~t1.bool_const).count() == 2
+        res = t1.collect()
+        assert res['bool_const'] == [False, False]
+        t1.insert([{'c1': 3}, {'c1': 4}])
+        assert t1.where(~t1.bool_const).count() == 4
+        res = t1.collect()
+        assert res['bool_const'] == [False, False, False, False]
+
+        # test adding a bool column with constant value to a view
+        t2 = pxt.create_table('test2', {'c1': pxt.Int})
+        validate_update_status(t2.insert([{'c1': 1}, {'c1': 2}]), expected_rows=2)
+        v = pxt.create_view('test_view', t2)
+        assert v.count() == 2
+        v.add_column(bool_const=True)
+        assert v.where(v.bool_const).count() == 2
+        res = v.collect()
+        assert res['bool_const'] == [True, True]
+        t2.insert([{'c1': 3}, {'c1': 4}])
+        assert v.where(v.bool_const).count() == 4
+        res = v.collect()
+        assert res['bool_const'] == [True, True, True, True]
+
+        # test using the bool column in a conditional expression
+        res = v.select((v.c1 > 1) & v.bool_const).collect()
+        assert len(res) == 4
+        assert res['col_0'] == [False, True, True, True]
+        # reversing the condition order should not affect the result
+        res = v.select(v.bool_const & (v.c1 > 1)).collect()
+        assert len(res) == 4
+        assert res['col_0'] == [False, True, True, True]
+
+        # sanity test persistence
+        _ = reload_tester.run_query(t1.select())
+        _ = reload_tester.run_query(t2.select())
+        _ = reload_tester.run_query(v.select())
+
+        # ... restart python session ...
+        _ = reload_tester.run_reload_test(clear=True)
+
+        t1 = pxt.get_table('test1')
+        t2 = pxt.get_table('test2')
+        v = pxt.get_table('test_view')
+        assert t2.count() == 4
+        assert v.count() == 4
+        res = v.collect()
+        assert res['bool_const'] == [True, True, True, True]
```

Comment:
redundant, the reload_tester already did that

Reply:
Done.
<!-- Reviewable comment -ODjhNWF7m1jfSgEgwoO:b-896fix -->
<!-- Sent from Reviewable.io -->